python/lsst/ap/association/association.py

```diff
@@ -114,30 +114,39 @@ def run(self,
               not matched a new DiaSource. (`int`)
         """
         diaSources = self.check_dia_source_radec(diaSources)
+
+        if self.config.doTrailedSourceFilter:
+            diaTrailedResult = self.trailedSourceFilter.run(diaSources,
+                                                            exposure_time)
+            diaSources = diaTrailedResult.diaSources
+            trailedSources = diaTrailedResult.trailedDiaSources
+
+            self.log.info(
+                "%i DIASources exceed max_trail_length, dropping fromsource catalog." % len(
+                    diaTrailedResult.trailedDiaSources))
+            self.metadata.add("num_filtered",
+                              len(diaTrailedResult.trailedDiaSources))
+        else:
+            trailedSources = pd.DataFrame(columns=diaSources.columns)
+
         if len(diaObjects) == 0:
             return pipeBase.Struct(
                 matchedDiaSources=pd.DataFrame(columns=diaSources.columns),
                 unAssocDiaSources=diaSources,
                 nUpdatedDiaObjects=0,
-                nUnassociatedDiaObjects=0)
+                nUnassociatedDiaObjects=0,
+                longTrailedSources=trailedSources)
 
-        if self.config.doTrailedSourceFilter:
-            diaTrailedResult = self.trailedSourceFilter.run(diaSources, exposure_time)
-            matchResult = self.associate_sources(diaObjects, diaTrailedResult.diaSources)
-
-            self.log.info("%i DIASources exceed max_trail_length, dropping "
-                          "from source catalog." % len(diaTrailedResult.trailedDiaSources))
-
-        else:
-            matchResult = self.associate_sources(diaObjects, diaSources)
+        matchResult = self.associate_sources(diaObjects, diaSources)
```

Comment:
Was this implemented wrong before and your commit is fixing a bug? Just trying to understand

Reply:
This wasn't implemented wrong per say, but trailed sources would not be run on a diaSource catalog if there were no diaObjects. This meant there were situation where the filtering was skipped, so I moved things around so it would always run and the object check would be on an already filtered diaSource catalog.